Assets/Mirror/Tests/Editor/CompressionFloatTest.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+
+namespace Mirror.Tests
+{
+    public class CompressionFloatTest
+    {
+        [Test]
+        [TestCaseSource(nameof(FloatTestCases))]
+        public void CanScaleToUintAndBack(float value, float minFloat, float maxFloat, uint minUint, uint maxUint, float allowedPrecision)
+        {
+            uint packed = Compression.ScaleToUInt(value, minFloat, maxFloat, minUint, maxUint);
+            float unpacked = Compression.ScaleFromUInt(packed, minFloat, maxFloat, minUint, maxUint);
+
+            Assert.That(unpacked, Is.Not.NaN);
+            Assert.That(unpacked, Is.EqualTo(value).Within(allowedPrecision), $"Value not in Allowed Precision\n   value    : {value}\n   unpacked: {unpacked}");
+        }
+
+        static IEnumerable FloatTestCases
```

Comment:
This is just uses the test frame work to test multiple values. There is no point testing functions like this with only 1 value because they doesn't cover (positive/negative/different precision/etc). There are 2 options, either write 100s of tests to cover common cases that could come up, or write 1 test and pass in different augments.

Reply:
Looking for examples of using `TestCaseSource` this is a common example of how to use them

```cs
  public static IEnumerable TestCases
  {
    get
    {
      yield return new TestCaseData( 12, 3 ).Returns( 4 );
      yield return new TestCaseData( 12, 2 ).Returns( 6 );
      yield return new TestCaseData( 12, 4 ).Returns( 3 );
      yield return new TestCaseData( 0, 0 )
        .Throws(typeof(DivideByZeroException))
        .SetName("DivideByZero")
        .SetDescription("An exception is expected");
    }
  }
```